src/Cmgmyr/Messenger/Models/Message.php

```diff
@@ -94,4 +94,25 @@ public function recipients()
     {
         return $this->participants()->where('user_id', '!=', $this->user_id);
     }
+
+    /**
+     * Prepares Builder for unread user messages
+     *
+     * @param $query
+     * @param $userId
+     * @return mixed
+     */
+    public function scopeUnreadIncomingForUser($query, $userId)
+    {
+
+        return $query->has('thread')
+            ->where('user_id', '!=', $userId)
+            ->whereHas('participants', function ($query) use ($userId) {
+
+                $query->where('user_id', $userId)
+                    ->where('last_read', '<', \DB::raw($this->getTable() . '.created_at'));
```

Comment:
Please import `DB` at the top of the file

Reply:
@nikolaynesov `scopeUnreadForUser` works, good call

Reply:
Ok, thanks. Let me know if we need any other updates.